pkg/occlient/occlient_test.go

```diff
@@ -1952,6 +1973,227 @@ func TestNewAppS2I(t *testing.T) {
+										"tags":     "builder,nodejs",
+									},
+									From: &corev1.ObjectReference{
+										Kind: "ImageStreamTag",
+										Name: "latest",
+									},
+									Name: "latest",
+								},
+							},
+						},
+					}
+	*/
+	tests := []struct {
+		name      string
+		imageNS   string
+		imageName string
+		imageTag  string
+		wantErr   bool
+		want      *imagev1.ImageStream
+	}{
+		{
+			name:      "Case: Valid request for imagestream of latest version and not namespace qualified",
+			imageNS:   "",
+			imageName: "foo",
+			imageTag:  "latest",
+			want:      fakeImageStream("foo", "testing", []string{"latest"}),
+		},
+		{
+			name:      "Case: Valid explicit request for specific namespace qualified imagestream of specific version",
+			imageNS:   "openshift",
+			imageName: "foo",
+			imageTag:  "latest",
+			want:      fakeImageStream("foo", "openshift", []string{"latest", "3.5"}),
+		},
+		{
+			name:      "Case: Valid request for specific imagestream of specific version not in current namespace",
+			imageNS:   "",
+			imageName: "foo",
+			imageTag:  "3.5",
+			want:      fakeImageStream("foo", "openshift", []string{"latest", "3.5"}),
+		},
+		{
+			name:      "Case: Valid request for specific imagestream of specific version not in current namespace",
+			imageNS:   "",
+			imageName: "foo",
+			imageTag:  "3.5",
+			want:      fakeImageStream("foo", "openshift", []string{"latest", "3.5"}),
+		},
+		{
+			name:      "Case: Invalid request for non-current and non-openshift namespace imagestream/Non-existant imagestream",
+			imageNS:   "foo",
+			imageName: "bar",
+			imageTag:  "3.5",
+			wantErr:   true,
+		},
+		{
+			name:      "Case: Request for non-existant tag",
+			imageNS:   "",
+			imageName: "foo",
+			imageTag:  "3.6",
+			wantErr:   true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fkclient, fkclientset := FakeNew()
+			fkclient.namespace = "testing"
+			openshiftIS := fakeImageStream(tt.imageName, "openshift", []string{"latest", "3.5"})
+			currentNSIS := fakeImageStream(tt.imageName, "testing", []string{"latest"})
+
+			fkclientset.ImageClientset.PrependReactor("get", "imagestreams", func(action ktesting.Action) (bool, runtime.Object, error) {
+				if tt.imageNS == "" {
+					if isTagInImageStream(*fakeImageStream("foo", "testing", []string{"latest"}), tt.imageTag) {
+						return true, currentNSIS, nil
+					} else if isTagInImageStream(*fakeImageStream("foo", "openshift", []string{"latest", "3.5"}), tt.imageTag) {
+						return true, openshiftIS, nil
+					}
+					return true, nil, fmt.Errorf("Requested imagestream %s with tag %s not found", tt.imageName, tt.imageTag)
+				}
+				if tt.imageNS == "testing" {
+					return true, currentNSIS, nil
+				}
+				if tt.imageNS == "openshift" {
+					return true, openshiftIS, nil
+				}
+				return true, nil, fmt.Errorf("Requested imagestream %s with tag %s not found", tt.imageName, tt.imageTag)
+			})
+
```

Comment:
Also check for `fkclientset.ImageClientset.Actions`

Reply:
Aah commented it out while debugging something.
Will un-comment the test

Reply:
Ack